var/spack/repos/builtin/packages/py-beautifulsoup4/package.py

```diff
@@ -25,11 +26,13 @@ class PyBeautifulsoup4(PythonPackage):
     variant("lxml", default=False, description="Enable lxml parser")
     variant("html5lib", default=False, description="Enable html5lib parser")
 
-    depends_on("python@3.6:", when="@4.11.0:", type=("build", "run"))
-    depends_on("python@3:", when="@4.10.0:", type=("build", "run"))
-    depends_on("py-setuptools", type="build")
+    depends_on("py-hatchling", when="@4.12.1:", type="build")
+    depends_on("py-setuptools", when="@:4.12.0", type="build")
+
     depends_on("py-soupsieve@1.3:", when="@4.9.0:", type=("build", "run"))
     depends_on("py-soupsieve@1.2:", when="@4.7.0:", type=("build", "run"))
 
     depends_on("py-lxml", when="+lxml", type=("build", "run"))
     depends_on("py-html5lib", when="+html5lib", type=("build", "run"))
+
+    depends_on("py-pytest", type="test")
```

Comment:
We don't actually run any tests, so you should prob remove this dep

Reply:
Without this dep installation with `--test=root` fails. I thought that `py-pytest` would be an automatic dependency when running `--test` but it is not. Is there a problem when adding it as test dependency? I thought this would be the use case for it?

Reply:
It's kind of complicated. `spack install --test=root` and `spack test run` both run our import tests, and any additional tests that users add to the package. However, `spack install --test=root` adds test dependencies while `spack test run` only adds run-time dependencies (@tldahlgren correct me if that changed). I guess I'm fine with adding it as a test dependency, as long as it isn't a run-time dependency too.